### PR TITLE
Issue #209: Merge parsers and update problem sets

### LINKED ISSUE #209: Merge existing FitBenchmarking and SASView problem formats
With #199 a new format problem definition file format is added as described in benchmark_problems/README.md

A longer term concern I have, is if a new problem definition file format needs to be added for every new software FitBenchmarking support since sounds like recipe for a longer term maintenance problem. 

It this issue, as a follow up to #199, investigate options for merging the FitBenchmark and sasview problem definition formats into a new format that can be used to describe both while being accommodable to support of other software in the future

====== **EDIT** (26/11/19) ==================
After the refactor of how parsing is done (#264), both of these parsers show very similar code.

This will require a new variable in the format to select which software to use for the function evaluation.

### PR DESCRIPTION
**Pull request by AndrewLister-STFC**
#### Description of Work

Fixes #209 

Removes sasview parser and merges functionality into fitbenchmarking parser

#### Testing Instructions

1. Run some of the changed files and check that output is the same as master for various solvers

Function: Does the change do what it's supposed to?
Tests: Does it pass? Is there adequate coverage for new code?
Style: Is the coding style consistent? Is anything overly confusing?
Documentation: Is there a suitable change to documentation for this change?

### CODE REVIEW

File: fitbenchmarking/parsing/fitbenchmark_parser.py [hunk 1]
```diff
@@ -29,35 +45,64 @@ def parse(self):
...
+
         self._parsed_func = self._parse_function()
 
+        # NAME
         fitting_problem.name = self._entries['name']
 
-        data_points = self._get_data_points()
+        # DATA
+        if software == 'mantid':
+            data_points = self._get_mantid_data_points()
```
Comment (wathen) [inline fitbenchmarking/parsing/fitbenchmark_parser.py]: Can we merge the `get_data_points` for both mantid and sasview?

Comment (AndrewLister-STFC) [reply fitbenchmarking/parsing/fitbenchmark_parser.py]: done

File: examples/benchmark_problems/Neutron/ENGINX236516_vanadium_bank1_10brk.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # FitBenchmark Problem
-# An example of fitting of a spline background on a Vanadium run from ENGIN-X
+software = 'Mantid'
 name = 'ENGINX 236516 vanadium, bank 1, 10 brk'
+description = 'An example of fitting of a spline background on a Vanadium run from ENGIN-X'
 input_file = 'ENGINX236516_vanadium_bank1.txt'
-function = 'name=BSpline, Order=3, NBreak=10, StartX=0.0, EndX=5.4'
-fit_parameters = {'StartX': 0, 'EndX': 5.4}
-description = ''
+function = 'name=BSpline, Order=3, NBreak=10'
```
Comment (wathen) [inline examples/benchmark_problems/Neutron/ENGINX236516_vanadium_bank1_10brk.txt]: It seems that the [bspline](https://www.mantidproject.org/BSpline) function in mantid needs to have `StartX` and `EndX` as attributes. 
From setting the function to `function = 'name=BSpline, Order=3, NBreak=10, StartX=0.0, EndX=5.4'` in the branch gives (which are the same as master):
with
and without gives:
without

Comment (wathen) [inline examples/benchmark_problems/Neutron/ENGINX236516_vanadium_bank1_10brk.txt]:
```suggestion
function = 'name=BSpline, Order=3, NBreak=10, StartX=0.0, EndX=5.4'
```

File: examples/benchmark_problems/Neutron/ENGINX236516_vanadium_bank1_20brk.txt [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # FitBenchmark Problem
-# An example of fitting of a spline background on a Vanadium run from ENGIN-X
+software = 'Mantid'
 name = 'ENGINX 236516 vanadium, bank 1, 30 brk'
+description = 'An example of fitting of a spline background on a Vanadium run from ENGIN-X'
 input_file = 'ENGINX236516_vanadium_bank1.txt'
-function = 'name=BSpline, Order=3, NBreak=20, StartX=0.0, EndX=5.4'
-fit_parameters = {'StartX': 0, 'EndX': 5.4}
-description = ''
+function = 'name=BSpline, Order=3, NBreak=20'
```
Comment (wathen) [inline examples/benchmark_problems/Neutron/ENGINX236516_vanadium_bank1_20brk.txt]:
```suggestion
function = 'name=BSpline, Order=3, NBreak=20, StartX=0.0, EndX=5.4'
```

**Pull request merged**